File: api/v2010/incoming_phone_number/update-example-1/update-example-1.5.x.rb
# Download the helper library from https://www.twilio.com/docs/ruby/install
require 'rubygems'
require 'twilio-ruby'

# Your Account Sid and Auth Token from twilio.com/console
account_sid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX'
auth_token = 'your_auth_token'
@client = Twilio::REST::Client.new(account_sid, auth_token)

incoming_phone_number = @client
  .incoming_phone_numbers('PNXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
  .update(
     account_sid: 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX',
     sms_url: 'http://demo.twilio.com/docs/sms.xml',
     voice_url: 'http://demo.twilio.com/docs/voice.xml'
   )

puts incoming_phone_number.account_sid

File: api/v2010/incoming_phone_number/update-example-1/update-example-1.5.x.php
<?php

// Update the path below to your autoload.php,
// see https://getcomposer.org/doc/01-basic-usage.md
require_once '/path/to/vendor/autoload.php';

use Twilio\Rest\Client;

// Find your Account Sid and Auth Token at twilio.com/console
$sid    = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
$token  = "your_auth_token";
$twilio = new Client($sid, $token);

$incoming_phone_number = $twilio->incomingPhoneNumbers("PN2a0747eba6abf96b7e3c3ff0b4530f6e")
                                ->update(array(
                                             "accountSid" => "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
                                             "smsUrl" => "http://demo.twilio.com/docs/sms.xml",
                                             "voiceUrl" => "http://demo.twilio.com/docs/voice.xml"
                                         )
                                );

print($incoming_phone_number->friendlyName);